api/src/shared/mail/infrastructure/services/smtp-mailer.js

```diff
@@ -0,0 +1,29 @@
+import { createTransport } from 'nodemailer';
+import { config } from '../../../config.js';
+
+export class SmtpMailer {
+  #transporter;
+
+  constructor() {
+    this.#transporter = SmtpMailer.createTransport();
+  }
+
+  static createTransport() {
+    return createTransport(config.mailing.smtpUrl);
+  }
```

Comment:
ça ne serait pas plus propre de passer les vars d'env nécessaires via le constructor ?
Voir [cet article](https://notestoself.dev/posts/code-smell-env-vars-in-constructor/#:~:text=Config%20is%20a%20dependency%20like%20any%20other%2C%20and%20should%20be%20injected%20into%20the%20class%20constructor.).

Reply:
En effet, mais quelle serait l'alternative ? Si on passe `smtpUrl` en argument au constructeur, j'ai l'impression qu'on ne fait que décaler le problème, la "variable globale" qu'est la var d'env sera toujours importée, mais ailleurs. Et potentiellement à plusieurs endroits, puisqu'il faudra le faire à chaque fois qu'on instancie la classe SmtpMailer.

Reply:
Ah oui c'est parce que j'ai l'habitude d'avoir des instanciations de services dans le cadre d'injections de dépendance. Dans ce cas-là, toutes les injections se font à la racine du projet (grossomodo) et puis nul part ailleurs

Reply:
Ok, du coup une solution serait d'avoir un service "config" récupérable via l'injection de dépendance et qu'on passerait au service depuis le usecase ?

Reply:
Disons qu'à l'endroit où les injections se font, on aurait tendance à instancier le SmtpMailer en y injectant la config, puis on injecte l'instance du SmtpMailer dans le ou les usecases qui en ont besoin.
Mais je ne sais pas si notre méthode d'injection permet cette souplesse.

Reply:
Mais en vrai ce bazar ça sert surtout quand tu as des tokens ou des url du provider pour la prod, le staging, les tests, docker etc. et que tu veux faire du e2e en décidant de quel env. tu veux utiliser